src/react/databrowser/objects/__tests__/Metadata.test.jsx

```diff
@@ -0,0 +1,137 @@
+import * as s3objects from '../../../actions/s3object';
+import { METADATA_SYSTEM_TYPE, METADATA_USER_TYPE } from '../../../utils';
+import { Item } from '../../../ui-elements/EditableKeyValue';
+import Metadata from '../details/Metadata';
+import { OBJECT_METADATA } from '../../../actions/__tests__/utils/testUtil';
+import React from 'react';
+import { reduxMount } from '../../../utils/test';
+
+describe('Metadata', () => {
+    const putObjectMetadataMock = jest.spyOn(s3objects, 'putObjectMetadata').mockReturnValue({ type: '' });
+    const optionLabels = ['cache-control', 'content-disposition', 'content-encoding', 'content-type', 'website-redirect-location', 'x-amz-meta', 'content-language', 'expires'];
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('Metadata should render', () => {
+        const { component } = reduxMount(<Metadata objectMetadata={OBJECT_METADATA}/>);
+
+        expect(component.find(Metadata).isEmptyRender()).toBe(false);
+    });
+
+    it('should render by default a SelectBox with empty values in each input when there are no key/value present', () => {
+        const { component } = reduxMount(<Metadata objectMetadata={OBJECT_METADATA}/>);
+
+        // check if only one item row is rendered
+        const items = component.find(Item);
+        expect(items).toHaveLength(1);
+
+        // check if item row values are rendered
+        const firstItem = items.first();
+        expect(firstItem.find('input[name="mdKeyType"]').prop('value')).toBe('');
+        expect(firstItem.find('input#metadata-input-value').prop('value')).toBe('');
+    });
+
+    it('should add new key/value metadata and should trigger function if save button is pressed', () => {
+        const { component } = reduxMount(<Metadata objectMetadata={OBJECT_METADATA}/>);
+
+        // check if only one item row is rendered
+        expect(component.find(Item)).toHaveLength(1);
+
+        // open dropdown menu
+        const inputComponent = component.find('input').first();
+        inputComponent.simulate('change');
+
+        // check all options are present
+        const optionsComponents = component.find('Option');
+        optionsComponents.forEach((optionComponent, index) => {
+            expect(optionComponent.find('div.sc-select-option-label').text()).toBe(optionLabels[index]);
+        });
+
+        // select a key
+        const firstItem = optionsComponents.first();
+        firstItem.simulate('click');
+
+        // check dropdown menu disappear
+        expect(component.find('Option')).toHaveLength(0);
```

Comment:
We can use `firstItem` instead of `component` to make sure we are in the first `Item`. 
Same thing for the following assertions although there is only one here.

Reply:
done